geopandas/_vectorized.py

```diff
@@ -200,7 +201,7 @@ def from_wkt(data):
     out = []
 
     for geom in data:
-        if geom is not None and len(geom):
+        if pd.notna(geom) and len(geom):
```

Comment:
A bit higher up in this file is a `_isna` function defined (as a scalar version of `pd.isna/notna`). 
Now, that doesn't handle pd.NA, but you could update that to do so, and then use that here instead of the `is not None` check.

Reply:
Actually, `array.py`, there is already a version of this that handles pd.NA:

https://github.com/geopandas/geopandas/blob/910131ec3af6143ccac33de4b66e9f8267c6fc7c/geopandas/array.py#L59-L73

So you copy that implementation here (into `_vectorized.py`) to use the same logic (and to avoid duplication, in array.py we can use the version defined in `_vectorized.py`)

Reply:
I'm not sure I understand the reasoning for using the scalar version on each element of a Series, rather than using the Pandas version on the entire Series as a filter.

Reply:
Because the way the current code here is written, is using a for loop over each element (so inside the for loop, we are dealing with scalar geometries). 

That's certainly something that could be optimized, but that's a bigger change than fixing the NA handling ;)